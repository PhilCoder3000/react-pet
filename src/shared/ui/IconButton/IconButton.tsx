import { ButtonHTMLAttributes, MouseEvent, PropsWithChildren } from 'react';
import { classnames } from 'shared/utils/classnames/classnames';
import classes from './IconButton.module.scss';

type Color = 'primary' | 'secondary';

interface IconButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  color?: Color;
  onClick?: (e: MouseEvent<HTMLButtonElement>) => void;
}

export function IconButton({
  children,
  color = 'primary',
  ...otherProps
}: IconButtonProps) {
  return (
    <button
      className={classnames(classes.button, classes[color])}
      {...otherProps}
    >
      {children}
    </button>
  );
}
