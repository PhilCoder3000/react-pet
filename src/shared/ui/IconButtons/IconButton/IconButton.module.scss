.button {
  border: none;
  border-radius: 50%;
  background: transparent;
  display: flex;
  width: 40px;
  height: 40px;
  justify-content: center;
  align-items: center;
  transition: background-color 200ms;
}

.primary {
  background-color: var(--bg-primary);
  & > * {
    color: var(--text-primary);
    fill: var(--text-primary);
  }
  &:hover {
    background-color: var(--bg-primary-hover);
  }
}

.secondary {
  background-color: var(--bg-secondary);
  & > * {
    color: var(--text-secondary);
    fill: var(--text-secondary);
  }
  &:hover {
    background-color: var(--bg-secondary-hover);
  }
}