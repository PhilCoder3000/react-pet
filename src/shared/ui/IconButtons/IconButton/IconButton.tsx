import { ButtonHTMLAttributes, MouseEvent } from 'react';
import { classnames } from 'shared/utils/classnames/classnames';
import classes from './IconButton.module.scss';

type Color = 'primary' | 'secondary';

export interface IconButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  color?: Color;
  onClick?: (e: MouseEvent<HTMLButtonElement>) => void;
}

export function IconButton({
  children,
  color = 'primary',
  className,
  ...otherProps
}: IconButtonProps) {
  return (
    <button
      data-testid="icon-button"
      {...otherProps}
      className={classnames(classes.button, classes[color], className)}
    >
      {children}
    </button>
  );
}
