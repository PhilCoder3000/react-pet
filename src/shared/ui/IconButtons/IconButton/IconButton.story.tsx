import type { ComponentStory, ComponentMeta } from '@storybook/react';
import { primaryBgDecorator } from 'shared/utils/storybookDecorators/PrimaryBg';
import { IconButton } from './IconButton';

export default {
  title: 'shared/IconButtons/IconButton',
  component: IconButton,
  args: {
    children: <span />
  }
} as ComponentMeta<typeof IconButton>;

const Template: ComponentStory<typeof IconButton> = (args) => (
  <IconButton {...args} />
);

export const Primary: ComponentStory<typeof IconButton> = Template.bind({});
Primary.args = {
  color: 'primary',
};

export const Secondary: ComponentStory<typeof IconButton> = Template.bind({});
Secondary.args = {
  color: 'secondary',
};
Secondary.decorators = [primaryBgDecorator]
