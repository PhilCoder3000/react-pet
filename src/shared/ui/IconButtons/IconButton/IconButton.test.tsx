import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { IconButton } from './IconButton';

const testId = 'icon-button';

describe('IconButton', () => {
  test('Render', () => {
    const { getByTestId } = render(<IconButton />);

    expect(getByTestId(testId)).toBeInTheDocument();
    expect(getByTestId(testId)).toBeEmptyDOMElement();
    expect(getByTestId(testId)).toBeEnabled();
  });

  test('Add class', () => {
    const { getByTestId } = render(<IconButton className="class" />);

    expect(getByTestId(testId)).toHaveClass('class');
  });

  test('Add value', () => {
    const { getByTestId } = render(<IconButton>value</IconButton>);

    expect(getByTestId(testId)).toHaveTextContent('value');
  });

  test('Add inner element', () => {
    const { getByTestId } = render(
      <IconButton>
        <span data-testid="children">span</span>
      </IconButton>,
    );
    const children = getByTestId('children');

    expect(getByTestId(testId)).toContainElement(children);
  });

  test('Disabled', () => {
    const { getByTestId } = render(<IconButton disabled />);

    expect(getByTestId(testId)).toBeDisabled();
  });

  test('Snapshot primary', () => {
    const snapshot = renderer.create(<IconButton color="primary" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  test('Snapshot secondary', () => {
    const snapshot = renderer.create(<IconButton color="secondary" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
