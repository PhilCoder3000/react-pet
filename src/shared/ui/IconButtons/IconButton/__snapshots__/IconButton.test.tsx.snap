// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton Snapshot primary 1`] = `
<button
  className="button primary"
  data-testid="icon-button"
/>
`;

exports[`IconButton Snapshot secondary 1`] = `
<button
  className="button secondary"
  data-testid="icon-button"
/>
`;
